h2/src/main/org/h2/mvstore/tx/Transaction.java

```diff
@@ -190,25 +198,54 @@ public int getStatus() {
         return getStatus(statusAndLogId.get());
     }
 
-    BitSet getCommittingTransactions() {
-        if (committingTransactions == null) {
-            return store.committingTransactions.get();
+    /**
+     * Get the root reference for the given map id.
+     *
+     * @param mapId the map id
+     * @return the root reference
+     */
+    Snapshot getMapRoot(int mapId) {
+        Snapshot snapshot = mapRoots.get(mapId);
+        if (snapshot == null) {
+            snapshot = getNewCurrentMapRoot(mapId);
         }
-        return committingTransactions;
+        return snapshot;
     }
 
     /**
-     * Get the root reference for the given map id.
+     * Get the current root reference from the start of the command for the
+     * given map id.
      *
      * @param mapId the map id
      * @return the root reference
      */
-    RootReference getMapRoot(int mapId) {
-        RootReference rootReference = mapRoots.get(mapId);
-        if (rootReference == null) {
-            rootReference = store.openMap(mapId).flushAndGetRoot();
+    Snapshot getCurrentMapRoot(int mapId) {
+        Snapshot snapshot = mapCurrentRoots.get(mapId);
+        if (snapshot == null) {
+            snapshot = getNewCurrentMapRoot(mapId);
         }
-        return rootReference;
+        return snapshot;
+    }
+
+    /**
+     * Get the new current root reference for the given map id.
+     *
+     * @param mapId the map id
+     * @return the root reference
+     */
+    Snapshot getNewCurrentMapRoot(int mapId) {
```

Comment:
Don't you think that createSnapshot would be more descriptive?

Reply:
Yes, I would be better, current name is complicated and obscure.